components/VM_Arm/src/main.c

```diff
@@ -882,7 +900,7 @@ static int load_vm(vm_t *vm, const char *kernel_name, const char *dtb_name, cons
         vm_ram_touch(vm, dtb_addr, size_gen, load_generated_dtb, gen_fdt);
         printf("Loading Generated DTB");
         dtb = dtb_addr;
-    } else {
+    } else if (vm_image_config.provide_dtb) {
```

Comment:
Does this mean that if generate_dtb or provide_dtb aren't set, then no dtb is loaded? Which would be fine for guests that don't require a dtb?

Reply:
Correct. I could add a ZF_LOG statement there to warn users. But specifically RTEMS doesn't need a DTB, which is why this was implemented.

Reply:
We might want to catch mis-configuration cases. What happen in a LInux VM if none of the filed is set (by accident or dev error). Is the DTB a NULL then and and there is a way to see what is wrong from the logs?

Reply:
Maybe add a ZF_LOGW?

Reply:
ZF_LOGW added to warn users of a potentially breaking state.